pennylane/devices/qubit/simulate.py

```diff
@@ -237,7 +254,189 @@ def simulate(
     tensor([0.68117888, 0.        , 0.31882112, 0.        ], requires_grad=True))
 
     """
-    state, is_state_batched = get_final_state(circuit, debugger=debugger, interface=interface)
+    has_mcm = has_mid_circuit_measurements(circuit)
+    has_shots = circuit.shots.total_shots is not None
+    if has_mcm and has_shots:
```

Comment:
I think maybe it makes sense to put everything inside this if block inside a different function